### ISSUE TITLE
sg run/start can operate against Bazel targets

### ISSUE THREAD

**Issue by jhchabran** [sourcegraph/sourcegraph#48195]

**Comment by burmudar** [comment]
Dave and I pair on this a bit. We discovered that on a fresh cold install when there are no binaries (nor folders for the binaries) a bunch of the commands that are supposed to built fail which makes sg quite entirely, not giving iBazel enough time to build any of the intended targets.

We've added a method `waitForBinary` method which takes the binLocation of a bazel command and waits for it to exist before returning. We can't use notify currently since the file / directory has to exist before hand and on a fresh install that won't exist.

The waiting is currently done naiively. We use a ticker and on every tick we use `os.Stat` to check if the file exists, if it does we stop the ticker and return.

The normal commands and the bazel commands get executed concurrently, but it can lead to some confusing log messages where the normal commands finish before the bazel ones, meaning the user will see "Sourcegraph is Ready" while in actual fact, iBazel has not finished compiling yet.

The next step is to have iBazel block sg while we wait for the commands to exists before continuiing.
